core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/TransactionSettingsTest.java

```diff
@@ -167,6 +167,32 @@ public void testDefaultOverride() {
 			assert transactionSettings.getValidationApproach() == ShaclSail.TransactionSettings.ValidationApproach.Auto;
 			assert !transactionSettings.isCacheSelectNodes();
 			assert !transactionSettings.isParallelValidation();
+			assert transactionSettings.getIsolationLevel() == IsolationLevels.SNAPSHOT_READ;
+
+			connection.commit();
+
+		}
+
+		sailRepository.shutDown();
+
+	}
+
+	@Test
+	public void testSerializableParallelValidation() {
+		ShaclSail shaclSail = new ShaclSail(new MemoryStore());
+		shaclSail.setParallelValidation(true);
+
+		SailRepository sailRepository = new SailRepository(shaclSail);
+		try (SailRepositoryConnection connection = sailRepository.getConnection()) {
+
+			connection.begin(IsolationLevels.SERIALIZABLE,
+					ShaclSail.TransactionSettings.PerformanceHint.ParallelValidation);
+
+			ShaclSailConnection sailConnection = (ShaclSailConnection) connection.getSailConnection();
+			ShaclSailConnection.Settings transactionSettings = sailConnection.getTransactionSettings();
+
+			assert transactionSettings.getValidationApproach() == ShaclSail.TransactionSettings.ValidationApproach.Auto;
+			assert !transactionSettings.isParallelValidation();
```

Comment:
Wrong kind of assert. Should be using junit `assertTrue` (or even better, AssertJ `assertThat`) here.